src/server/project.ts

```diff
@@ -380,9 +387,39 @@ namespace ts.server {
                 }
                 result.push(asNormalizedPath(f.fileName));
             }
+            if (!excludeConfigFiles) {
+                const configFile = this.program.getCompilerOptions().configFile;
+                if (configFile) {
+                    result.push(asNormalizedPath(configFile.fileName));
+                    if (configFile.extendedSourceFiles) {
+                        for (const f of configFile.extendedSourceFiles) {
+                            result.push(asNormalizedPath(f));
+                        }
+                    }
+                }
+            }
             return result;
         }
 
+        hasConfigFile(configFilePath: NormalizedPath) {

```

Comment:
I would prefer `isConfigFile` -- it doesn't return whether the project has a config file, it returns whether the *parameter* is the path to one of the project's config files.